src/spartan/polynomial.rs

```diff
@@ -69,10 +100,13 @@ impl<Scalar: PrimeField> MultilinearPolynomial<Scalar> {
     self.Z.len()
   }
 
+  // NOTE: this is equivalent to Espresso/HyperPlonk's 'fix_last_variables' method
```

Comment:
Maybe remove this comment as it could cause confusion: Espresso/HyperPlonk came way after Spartan (the code below is inherited from Spartan) rather than the other way around :)

Reply:
This wins as best comment of the review.

Reply:
In fact, this was an internal note for internal clarity while fixing stuff, not to be send in a PR.